pimd/pim_zebra.c

```diff
@@ -855,27 +855,24 @@ void igmp_source_forward_start(struct pim_instance *pim,
 		}
 	}
 
-	result = pim_channel_add_oif(source->source_channel_oil,
-				     group->group_igmp_sock->interface,
-				     PIM_OIF_FLAG_PROTO_IGMP);
-	if (result) {
-		if (PIM_DEBUG_MROUTE) {
-			zlog_warn("%s: add_oif() failed with return=%d",
-				  __func__, result);
+	if (PIM_I_am_DR(pim_oif)) {
```

Comment:
if the IGMP report is received on a LAN, non-DR would need to add the OIF. PIM assert mechanism would take care of the forwarding.